tests/python_test.py

```diff
@@ -285,3 +285,12 @@ def test_write_resource_invalid_align_sector(
     with pytest.raises(ValueError):
         sanlock.write_resource(
             "ls_name", "res_name", disks, align=align, sector=sector)
+
+
+def test_write_resource_invalid_path(tmpdir, sanlock_daemon):
+    # Test parsing a resource which is not a list of tuples
+    disks = ["invalid resource tuple"]
+    with pytest.raises(ValueError) as e:
+        sanlock.write_resource("ls_name", "res_name", disks)
+    assert "invalid resource tuple" in repr(e.value)
```

Comment:
We want to test that the result of repr("invalid resource tuple"), which is 'invalid resource tuple' (not the single quote) is inside the error message.

This should be correct:

    invalid_tuple = "invalid"
    disks = [invalid_tuple]
    ...
    assert repr(invalid_tuple) in str(e.value)

Also what about checking non ascii value? This is likely to break with the current code, using:

    str_rep = PyString_AsString(rep);

Reply:
I think this PR should be rebased upon python3 porting PR:
1. PyString_AsString is replaced with PyUnicode_AsUTF8 in python 3 porting
2. We will be able to test different invalid resource path scenarios over py3 and py2